packages/app/src/cli/services/app/config/link.ts

```diff
@@ -118,7 +124,17 @@ export default async function link(options: LinkOptions, shouldRenderSuccess = t
     renderSuccessMessage(configFileName, remoteAppConfiguration.name, localApp)
   }
 
-  return configurationIncludingRemote
+  return condensedConfiguration
+}
+
+function condenseComplianceAndNonComplianceWebhooks(config: CurrentAppConfiguration): CurrentAppConfiguration {
+  const webhooksConfig = (config.webhooks as WebhooksConfig) ?? {}
+  const subscriptionsArr = webhooksConfig.subscriptions ?? []
+  if (subscriptionsArr.length === 0) return config
+
+  const condensedSubscriptionsArr = reduceWebhooks(subscriptionsArr)
+  webhooksConfig.subscriptions = condensedSubscriptionsArr
+  return config
```

Comment:
The assignment and overwriting the underlying value is pretty confusing

```suggestion
  if (!config?.webhooks?.subscriptions?.length) return config
  config.webhooks.subscriptions = reduceWebhooks(subscriptionsArr)
  return config
```

Reply:
You can't do that because you need to get the `subscriptionsArr` first, and config is not properly typed here so you need to do the initial cast.

This line does make sense though:
`config.webhooks.subscriptions = reduceWebhooks(subscriptionsArr)`

Reply:
ah I missed that var, you could just do `config.webhooks.subscriptions`? Which at that point does make sense to pull this out, but still simplifies things - with edit:

```js
const subscriptionsArr = config?.webhooks?.subscriptions as WebhooksConfig["subscriptions"]
if (!subscriptionsArr.length) return config
  config.webhooks.subscriptions = reduceWebhooks(subscriptionsArr)
  return config
```